relayer/hermes/hermes_commander.go

```diff
@@ -0,0 +1,204 @@
+			},
+			ConnectionHops: r.ChannelEnd.ConnectionHops,
+			Version:        r.ChannelEnd.Version,
+			PortID:         r.ChannelEnd.Remote.PortID,
+			ChannelID:      r.ChannelEnd.Remote.ChannelID,
+		})
+	}
+
+	return ibcChannelOutput, nil
+}
+
+func (c commander) ParseGetConnectionsOutput(stdout, stderr string) (ibc.ConnectionOutputs, error) {
+	var queryResult ConnectionQueryResult
+	if err := json.Unmarshal([]byte(stdout), &queryResult); err != nil {
+		return ibc.ConnectionOutputs{}, err
+	}
+
+	var outputs ibc.ConnectionOutputs
+	for _, r := range queryResult.Result {
+
+		var versions []*ibcexported.Version
+		for _, v := range r.ConnectionEnd.Versions {
+			versions = append(versions, &ibcexported.Version{
+				Identifier: v.Identifier,
+				Features:   v.Features,
+			})
+		}
+
+		outputs = append(outputs, &ibc.ConnectionOutput{
+			ID:       r.ConnectionID,
+			ClientID: r.ConnectionEnd.ClientID,
+			Versions: versions,
+			State:    r.ConnectionEnd.State,
+			Counterparty: &ibcexported.Counterparty{
+				ClientId:     r.ConnectionEnd.Counterparty.ClientID,
+				ConnectionId: r.ConnectionEnd.Counterparty.ConnectionID,
+				Prefix: types.MerklePrefix{
+					KeyPrefix: []byte(r.ConnectionEnd.Counterparty.Prefix),
+				},
+			},
+		})
+	}
+	return outputs, nil
+}
+
+func (c commander) ParseGetClientsOutput(stdout, stderr string) (ibc.ClientOutputs, error) {
+	var queryResult ClientQueryResult
+	if err := json.Unmarshal([]byte(stdout), &queryResult); err != nil {
+		return ibc.ClientOutputs{}, err
+	}
+
+	var clientOutputs []*ibc.ClientOutput
+	for _, r := range queryResult.ClientResult {
+		clientOutputs = append(clientOutputs, &ibc.ClientOutput{
+			ClientID: r.ClientID,
+			ClientState: ibc.ClientState{
+				ChainID: r.ChainID,
+			},
+		})
+	}
+
+	return clientOutputs, nil
+}
+
+func (c commander) Init(homeDir string) []string {
+	return nil
+}
+
+func (c commander) GetChannels(chainID, homeDir string) []string {
+	// the --verbose and --show-counterparty options are required to get enough information to correctly populate
+	// the path.
+	return []string{hermes, "--json", "query", "channels", "--chain", chainID, "--show-counterparty", "--verbose"}
+}
+
+func (c commander) GetConnections(chainID, homeDir string) []string {
+	return []string{hermes, "--json", "query", "connections", "--chain", chainID, "--verbose"}
+}
+
+func (c commander) GetClients(chainID, homeDir string) []string {
+	return []string{hermes, "--json", "query", "clients", "--host-chain", chainID}
+}
+
+func (c commander) StartRelayer(homeDir string, pathNames ...string) []string {
+	return []string{hermes, "--json", "start", "--full-scan"}
+}
+
+func (c commander) CreateWallet(keyName, address, mnemonic string) ibc.Wallet {
+	return NewWallet(keyName, address, mnemonic)
+}
+
+func (c commander) UpdatePath(pathName, homeDir string, filter ibc.ChannelFilter) []string {
+	// TODO: figure out how to implement this.
+	panic("implement me")
+}
```

Comment:
I don't know if this is required or not for hermes, the tests are passing without.